webapp/src/pages/ResponseList/components/ResponseListComponent.js

```diff
@@ -0,0 +1,494 @@
+        .then(resp => {
+            if (resp.statusCode === 201) {
+                const newResponse = {
+                ...this.state.selectedResponse,
+                tags: [...this.state.selectedResponse.tags, resp.response.data]
+                } 
+                const newResponses = this.state.responses.map(r => 
+                    r.response_id === this.state.selectedResponse.response_id  ? newResponse : r);
+                this.setState({
+                tagSelectorVisible: false,
+                selectedResponse: null,
+                filteredTags: [],
+                responses: newResponses
+                }, this.filterResponses);
+                }
+            else {
+                this.setState({
+                tagSelectorVisible: false,
+                error: resp.error
+            });
+            }
+        })
+    }
+  
+  confirmRemoveTag = () => {
+    const {selectedResponse, selectedTag} = this.state;
+
+    responsesService.removeTag(selectedResponse.response_id, selectedTag.id, this.props.event.id)
+    .then(resp => {
+      if (resp.statusCode === 204) {
+        const newResponse = {
+          ...selectedResponse,
+          tags: selectedResponse.tags.filter(t=>t.id !== selectedTag.id)
+        }
+        const newResponses = this.state.responses.map(r => 
+            r.response_id === selectedResponse.response_id ? newResponse : r);
+        this.setState({
+          responses: newResponses,
+          confirmRemoveTagVisible: false
+        }, this.filterResponses);
+      }
+      else {
+        this.setState({
+          error: resp.error,
+          confirmRemoveTagVisible: false
+        });
+      }
+    })
+    }
+
+    updateNameSearch = (event) => {
+        this.setState({nameSearch: event.target.value}, this.filterResponses);
+    }
+    
+    updateEmailSearch = (event) => {
+        this.setState({emailSearch: event.target.value}, this.filterResponses);
+    }
+
+    updateTagSearch = (id, event) => {
+        this.setState({tagSearch: event.value}, this.filterResponses);
+    }
+
+    getSearchTags(tags) {
+        const tag_options = tags.map(t => ({ value: t.name, label: t.name }));
+        return [{ value: "none", label: this.props.t("None") }, ...tag_options];
+    }
+
+    filterResponses = () => {
+        const { nameSearch, tagSearch, emailSearch } = this.state;
+        const filtered = this.state.responses.filter(r => {
+          let passed = true;
+          if (nameSearch) {
+            passed = r.user.toLowerCase().indexOf(nameSearch.toLowerCase()) > -1;
+          }
+          if (emailSearch && passed) {
+            passed = r.email.toLowerCase().indexOf(emailSearch.toLowerCase()) > -1;
+          }
+          if (tagSearch && passed && tagSearch !== "none") {
+            passed = r.tags.some(t => t.name === tagSearch);
+          }
+          return passed;
+        });
+        this.setState({ filteredResponses: filtered });
+      }
+
+    render() {
```

Comment:
Some minor comments about the UI: 
- The response link column seems to wide, maybe just change the header to "ID" and make it more narrow? The fact that its blue will hopefully indicate to people that it's a link better than how it was previously displayed
- Would be great to change the cursor to pointer when you hover over the tag badges to indicate they're clickable 
- Can you get the Assign button to be not so wide? :) 
- We do need to be able to filter by multiple tags as an "AND" which the page could previously do, see how it's done in the [review assignment](https://github.com/deep-learning-indaba/Baobab/blob/d7ccb42ef6047466b0bcf64c19960e6a058670a8/webapp/src/pages/reviewAssignment/components/ReviewAssignmentComponent.js#L195) page for example. But if you want to keep the controls consistent so it looks like the other search boxes then perhaps we need a new form component like [the Multi example from React Select](https://react-select.com/home) -  and good news is that we are already using React Select so it should be easy to add this! :) 
Otherwise the new UI is beautiful!! :D :D

Reply:
I couldn't work out how to get the assign button to look good - any pointers? The rest, however, is done :)